File: backend/src/main/java/com/matricula/modules/carrera/service/CarreraService.java
package com.matricula.modules.carrera.service;

import com.matricula.modules.carrera.dto.CarreraDTO;

import java.util.List;

public interface CarreraService {
    List<CarreraDTO> findAll();
    CarreraDTO findById(Integer id);
    List<CarreraDTO> findByFacultad(Integer facultadId);
    CarreraDTO create(CarreraDTO dto);
    CarreraDTO update(Integer id, CarreraDTO dto);
    void delete(Integer id);
}

File: frontend/src/services/carreraService.js
import api from './api';

export const carreraService = {
  getAll: () => api.get('/api/carreras'),
  getById: (id) => api.get(`/api/carreras/${id}`),
  getByFacultad: (facultadId) => api.get(`/api/carreras/facultad/${facultadId}`),
  create: (carrera) => api.post('/api/carreras', carrera),
  update: (id, carrera) => api.put(`/api/carreras/${id}`, carrera),
  delete: (id) => api.delete(`/api/carreras/${id}`),
};